New tests for blocked_actions_protection find that it doesn't activate when expected

leoccyao opened the pull request:
This pr started with me trying to write tests for the recently added blocked_actions_sleep functionality; however, upon finding the tests failing, I found that the api send_request() method was not returning "feedback_required" upon hitting a block (see https://github.com/instagrambot/instabot/issues/1087#issuecomment-542549855). Therefore, a second commit was added to reinstate the return value of "feedback_required" (which was present in an earlier version, but seemed to have been removed somewhere).

leoccyao committed:
Wrote tests for blocked_actions_sleep

leoccyao committed:
api.py send_request return "feedback_required"

Before this change, a feedback_required response in send_request
would return False, therefore not triggering the check for
feedback_required in the like() call. This makes a detected
feedback_required explicitly return "feedback_required", triggering
the check.

leoccyao committed:
Code style issues

leoccyao committed:
More code style issues

ohld committed:
Merge branch 'master' into master

ohld commented on instabot/api/api.py:
```diff
@@ -454,6 +454,7 @@ def send_request(
                         "ATTENTION!: `feedback_required`"
                         + str(response_data.get("feedback_message"))
                     )
+                    return "feedback_required"
```
Why have you added it back? This is bug and should be removed.

ohld replied:
You shouldn’t include back this line!

leoccyao committed:
Encode feedback message in utf-8

leoccyao committed:
Even more code style issues...

leoccyao committed:
No need to utf-8 response message

Fixes AttributeErrors raised by other tests

leoccyao committed:
The code style is getting a bit annoying...

ohld reviewed (changes requested):
Please read my comments. Mind this PR: https://github.com/instagrambot/instabot/pull/1083

leoccyao committed:
Will not return feedback_required if two factor

ohld commented on instabot/api/api.py:
```diff
@@ -449,11 +449,15 @@ def send_request(
+                            "feedback_message").encode('utf-8'))
                     )
+                    if not response_data.get("two_factor_required"):
+                        return "feedback_required"
```
hmmm, the same "feedback_required" here. The same problem as we had this similar return value. Please prove that this return value is important (please take into account the pull request mentioned above) or just delete it.

ohld committed:
Merge branch 'master' into master

leoccyao committed:
Added utf-8 for response data message

leoccyao committed:
Added check_user option for follow

leoccyao committed:
Wrote blocked actions tests for follows

leoccyao committed:
Remove return feedback_required

ohld committed:
Merge branch 'master' into master

leoccyao committed:
Refactor two_factor_auth into separate method

This commit puts the two factor code into it's own method, and checks
if two factor is needed from both the send_request and the
login methods. Therefore, even if feedback_required is returned from the
send_request method, the second check in the login method will still
activate 2FA if needed.
